evalml/tests/component_tests/test_polynomial_detrender.py

```diff
@@ -104,10 +104,21 @@ def test_polynomial_detrender_needs_monotonic_index(ts_data):
         y_shuffled = y.sample(frac=1, replace=False)
         detrender.fit_transform(X, y_shuffled)
     assert "monotonically" in str(exec_info.value)
-
-    with pytest.raises(
-        NotImplementedError,
-        match="class 'pandas.core.indexes.base.Index'> is not supported",
-    ):
-        y_string_index = pd.Series(np.arange(31), index=[f"row_{i}" for i in range(31)])
-        detrender.fit_transform(X, y_string_index)
+    try:
+        with pytest.raises(
```

Comment:
sktime v0.10.0 raises a ValueError here, but our min_dependencies test raises a NotImplementedError instead since it uses sktime v0.7.0